WebAPIBookStore/WebAPIBookStore/Controllers/OrderController.cs

```diff
@@ -0,0 +1,123 @@
+                               IUserRepository userRepository,
+                               ICartItemRepository cartItemRepository,
+                               IMapper mapper)
+        {
+            _userRepository = userRepository;
+            _orderRepository = orderRepository;
+            _cartItemRepository = cartItemRepository;
+            _mapper = mapper;
+        }
+
+        [HttpGet]
+        public IActionResult GetOrders()
+        {
+            var orders = _orderRepository.GetOrders();
+            if (orders.Count <= 0)
+                return NotFound();
+            
+            var orderMaps = _mapper.Map<List<Order>>(orders);
+            return ModelState.IsValid? Ok(orderMaps) : BadRequest(ModelState);
+        }
+
+        [HttpGet("Status")]
+        public IActionResult GetOrderByStatus([FromQuery] string name)
+        {
+            var orders = _orderRepository.GetOrderByStatus(name);
+            if (orders.Count <= 0)
+                return NotFound();
+
+            var orderMaps = _mapper.Map<List<Order>>(orders);
+            return ModelState.IsValid ? Ok(orderMaps) : BadRequest(ModelState);
+        }
+
+        [HttpGet("{id}")]
+        public IActionResult GetOrder(int id)
+        {
+            var order = _orderRepository.GetOrder(id);
+            if (order == null)
+                return NotFound();
+
+            return ModelState.IsValid ? Ok(order) : BadRequest(ModelState);
+        }
+
+        [HttpGet("User/{userId}")]
+        public IActionResult GetOrderByUserId(int userId)
+        {
+            var orders = _orderRepository.GetOrderByUserId(userId);
+            if (orders.Count <= 0)
+                return NotFound();
+
+            return Ok(orders);
+        }
+
+        [HttpPost]
+        public IActionResult CreateOrder([FromBody] OrderCreate orderCreate)
+        {
+            if (!ModelState.IsValid)
+                return BadRequest(ModelState);
+
+            if (orderCreate.CartItemIds.Count <= 0)
+                return BadRequest(ModelState);
+
+            if (!_userRepository.UserExists(orderCreate.userId))
+                return NotFound("Not found user");
+
+            List<CartItem> cartItems = new List<CartItem>();
+            foreach (int id in orderCreate.CartItemIds)
+            {
+                var cartItem = _cartItemRepository.GetCartItem(id);
+                if (cartItem == null)
+                    return NotFound();
+
+                if (cartItem.Status == "Paid")
+                    return BadRequest();
+
+                cartItems.Add(cartItem);
+            }
+
+            var orderMap = _mapper.Map<Order>(orderCreate.OrderDto);
+            if (!_orderRepository.CreateOrder(cartItems, orderMap, orderCreate.userId))
+            {
+                ModelState.AddModelError("", "Something went wrong");
+                return StatusCode(500, ModelState);
+            }
+
+            return Ok("Successfully created");
```

Comment:
Update dùng toán tử 3 ngôi mà create lại return riêng như này à em? Chức năng gần tương tự nhau thì nên để format cũng như vậy. Thử tìm cách sửa cho hợp lý nhé!